### PR TITLE
allow using infile path in output.

### PR DESCRIPTION
**Pull request by satra**
@Hoda1394 - this should allow using the input name. but the README needs to be improved to indicate how to use this. 

i haven't actually tested this.

### CODE REVIEW

**Comment by Hoda1394** [comment]
I will modify the README.

**Comment by Hoda1394** [comment]
@satra - I tried to test it. it seems that it does not solve the multiple infiles issue if all infiles are in the same folder since we only give one prefix for all infiles.

**Comment by satra** [comment]
@Hoda1394 - ah indeed. we need to support using the name of the infile in the prefix and we were trying to avoid detecting what type of file it is to remove the extension. let me do one more thing.

**Comment by satra** [comment]
here is a bit of a variant. uses the infile name.

**Comment by Hoda1394** [comment]
yes, we need to use the infile name. I think, now it works.

File: kwyk/cli.py [hunk 1]
```diff
@@ -54,21 +55,26 @@ def predict(*, infiles, outprefix, model, n_samples, batch_size, save_variance,
...
+def _predict(infile, outprefix, predictor, n_samples, batch_size, save_variance, save_entropy, overwrite, atlocation, base):
     _orig_infile = infile
 
     # Are there other neuroimaging file extensions with multiple periods?
     if infile.lower().endswith('.nii.gz'):
         outfile_ext = '.nii.gz'
     else:
         outfile_ext = Path(infile).suffix
-    outfile_stem = outprefix
+    outfile_stem = outprefix + Path(infile).name.replace(outfile_ext, "")
```
Comment (kaczmarj) [inline kwyk/cli.py:69]: one can use `Path.stem` here.

```suggestion
    outfile_stem = outprefix + Path(infile).stem
```

Comment (satra) [reply kwyk/cli.py:69]: will it support both 'nii.gz' and 'nii' ?

Comment (kaczmarj) [reply kwyk/cli.py:69]: oops, it will not. nice catch.

**Comment by Hoda1394** [comment]
@satra - it is not clear to me what should the user enter as the value for the `base`? shouldn't it be a flag?

**Comment by satra** [comment]
if say the files are at some location `/path/to/location/bidsdataset/sub-1/anat/T1.nii.gz` the user could provide `--base /path/to/location/bidsdataset/`. this would remove that from the name. perhaps `base` is a terrible word for this operation. `--stripstr` may be a better keyword.

File: README.md [hunk 2]
```diff
@@ -31,14 +31,27 @@ singularity run -B $(pwd):/data -W /data --nv kwyk_latest-gpu.sif -m bvwn_multi_
...
+singularity run -B $(pwd):/data -W /data --nv kwyk_latest-gpu.sif -m bvwn_multi_prior -n 2 \
+  --save-variance --save-entropy path/to/T1_001.nii.gz path/to/T2_001.nii.gz output
+```
+
+If the input files share the same file path pattern such as bids format, the `--base` option removes parts of infile path and creates a new folder to save all outputs there.
+
+```
+singularity run -B $(pwd):/data -W /data --nv kwyk_latest-gpu.sif -m bvwn_multi_prior -n 2 \
+  --save-variance --save-entropy --base /path/to/location/bidsdataset/ /path/to/location/bidsdataset/sub-1/anat/T1.nii.gz output
+```
```
Comment (satra) [inline README.md:55]: it would be nice to have an example with multiple T1 files from a bidsdataset.

**Pull request merged**